File: env_variable/ruby_env_variable/env_variable.rb
#!/usr/bin/ruby -w
#comments

def env_variable_function(n)
  x = 0

  x = (n + n - n) * (n / 1)
  x = x * (1.25).floor

  if( x < 0 ) then
    x = x.abs
  end
  return x
end

def main()
  if ARGV.length != 0 then
    puts "Usage: #{$0} <noargs>"
    exit(1)
  end
  puts ENV['OS']
end

main()

File: env_variable/perl_env_variable/env_variable.pl
#!/usr/bin/perl -w

use strict;
use warnings;
use POSIX qw/floor/;
use POSIX qw/abs/;

sub main {
  my $idx_i;

  if( $#ARGV+1 != 0 ) {
    print STDERR "Usage: $0 <noargs>\n";
    exit(1);
  }
  print $ENV{OS}, "\n";
  exit(0);
}

sub env_variable_function {
  my $n = shift;
  my $x = 0;

  $x = ($n + $n - $n) * ($n / 1);
  $x = $x * floor(1.25);

  if( $x < 0 ) {
    $x = abs($x);
  }
  return $x;
}

&main();
